### ISSUE TITLE
[NSFW] xhamster: ads

### ISSUE THREAD

**Issue by Ziko577** [uBlockOrigin/uAssets#23538]
### Prerequisites

- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367), [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) or [_a shortener/hosting site_](https://github.com/uBlockOrigin/uAssets/discussions/17361) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] I read and understand the [policy about what is a valid filter issue](https://github.com/uBlockOrigin/uAssets/blob/master/README.md#uassets).
- [X] I verified that this issue is not a duplicate. (Search [here](https://github.com/uBlockOrigin/uAssets/issues) to find out.)
- [X] I did not remove any of the [default filter lists](https://github.com/uBlockOrigin/uAssets/assets/66517106/6e8d024a-5869-47cf-b142-b2357eb6c7d0), or I have verified that the issue was not caused by removing any of the default lists.
- [X] I did not enable additional filter lists, or I have verified that the issue still occurs without enabling additional filter lists.
- [X] I do not have custom filters/rules, or I have verified that the issue still occurs without custom filters/rules.
- [X] I am not using uBlock Origin along with other content blockers.
- [X] I have verified that the web browser's built-in content blocker/tracking protection, [network wide/DNS blocking](https://en.wikipedia.org/wiki/DNS_blocking), or my VPN is not causing the issue.
- [X] I have verified that other extensions are not causing the issue.
- [X] If this is about a breakage or detection, I have verified that it is caused by uBlock Origin and isn't a site issue.
- [X] I did not answer truthfully to **ALL** the above checkboxes.

### URL(s) where the issue occurs.

```yaml
https://xhamster.com/videos/yui-hatano-is-seen-jacking-while-performing-oral-fuck-a-thon-in-a-pornographic-video-titled-jav-xxx-xhUYTlR
https://xhamster.com/search/oriental+massage
```

### Description

Every 3rd video has a pesky ad that loads on top of the video and nothing I do seems to get rid of it. There's also the banner ad at the bottom and the top of the site as well as in between advertising FapHouse. Chat ads are also commonplace now even with picking them out they come back. 

### Other extensions used

None.

### Screenshot(s)

https://drive.google.com/file/d/1U9qGmKX3AyzuuLszonKLV2iy2vUNXmQy/view?usp=drive_link
https://drive.google.com/file/d/1SAUldXwngwEdno33B57wmS-9U1XOkV2X/view?usp=drive_link
https://drive.google.com/file/d/1De8gwesanmMlabOYLqvRjhCnfeWd4SG7/view?usp=drive_link

### Configuration

```
uBlock Origin: 1.57.0
Chromium: 109
filterset (summary):
 network: 135310
 cosmetic: 69
 scriptlet: 25507
 html: 0
listset (total-discarded, last-updated):
 added:
  https://gitlab.com/Sorrow-San/9anime-adblock-filters/raw/master/9Anime%20filter%20list: 148-33, 22h.15m
  ublock-annoyances: 6959-1843, 15m Δ
 default:
  user-filters: 98-6, never
  ublock-filters: 37810-10909, 15m Δ
  ublock-badware: 8505-678, 15m Δ
  ublock-privacy: 881-13, 15m Δ
  ublock-unbreak: 2268-332, 15m Δ
  ublock-quick-fixes: 262-151, 15m Δ
  easylist: 86841-35425, 15m Δ
  easyprivacy: 50847-816, 15m Δ
  urlhaus-1: 12895-0, 15m
  plowe-0: 3735-6, 2d.17h.57m
filterset (user): [array of 98 redacted]
trustedset:
 added: [array of 46 redacted]
switchRuleset:
 added: [array of 8 redacted]
hostRuleset:
 removed:
  behind-the-scene * * noop
  behind-the-scene * image noop
  behind-the-scene * 3p noop
  behind-the-scene * inline-script noop
  behind-the-scene * 1p-script noop
  behind-the-scene * 3p-script noop
  behind-the-scene * 3p-frame noop
userSettings:
 contextMenuEnabled: false
 ignoreGenericCosmeticFilters: true
 parseAllABPHideFilters: false
 showIconBadge: false
 userFiltersTrusted: true
hiddenSettings: [none]
supportStats:
 allReadyAfter: 12460 ms (selfie)
 maxAssetCacheWait: 1355 ms
 cacheBackend: indexedDB
```

**Comment by Zephiii** [comment]
> * [x] I did not answer truthfully to ALL the above checkboxes.

**Comment by Ziko577** [comment]
Thanks! This eliminated all of the ads completely once I removed the filters I made.

**Comment by AroKol78** [comment]
`/mlxhl/v1/for-page?$xhr` --> `/mlxhl/v2/for-page?$xhr`

I found `3p` more `?language=`
```
/tag_1?language=$xhr,3p
/category_1?language=$xhr,3p
/pornstar_2.html?language=$frame,3p
/channel_1.html?language=$frame,3p
```
maybe somehow in general `?language=$frame,xhr,3p` + cosmetics for holes left behind

**Comment by stephenhawk8054** [comment]
I don't see `/tag_1?` or `/category_1?` appear in the log. Can you give exact URL where you see those?

**Comment by AroKol78** [comment]
> I don't see `/tag_1?` or `/category_1?` appear in the log. Can you give exact URL where you see those?

<details><summary>screenshot log - example</summary>

![xhhhhh](https://github.com/uBlockOrigin/uAssets/assets/70756447/31d892f3-0ce0-4c7d-abe4-e7e5d1c01e39)

</details>

**Comment by ghost** [comment]
I noticed now that it if you click a certain number of times (any number) on a video suggested under the player, it wants (only once) to open a pop-up (with the site blocked), but then it doesn't happen again. 

https://github.com/user-attachments/assets/6f841d20-f034-4ef9-ac8e-0d36475e0d0b

**Comment by stephenhawk8054** [comment]
Which browser are you using? Mobile or desktop?

**Comment by ghost** [comment]
Firefox 141, on desktop. I should also mention that I use incognito window, maybe this also has an effect somehow.

**Comment by ghost** [comment]
I tried again two more times, first time, after opening two suggested videos, and second time, after 6 clicks on a suggested video. After these clicks the pop-up appears with the message like in the screenshot above.

Seems to be happening randomly, I first thought it needs a certain number of clicks for the popup to trigger, but it's not.

**Comment by stephenhawk8054** [comment]
Those popunders should be blocked by `xhamster.*##+js(trusted-replace-node-text, script, popunder, '', condition, popunder, stay, 1)`

Can you test these filters:

```adb
xhamster.*#@#+js(trusted-replace-node-text, script, popunder, '', condition, popunder, stay, 1)
xhamster.*##+js(trusted-replace-node-text, script, popunder, '')
```

Clear all the cookies to reproduce more easily.

Replace the domain in case you are redirecting to another xh domain.

**Comment by ghost** [comment]
In my case it's ro.xhamster.com. I added the filters, opened an incognito window, but it didn't work:
https://github.com/user-attachments/assets/2cb61a0a-055d-43ce-bdcf-7b8c814a8fb9

**Comment by stephenhawk8054** [comment]
You need to tick on `Allow custom filters requiring trust`

**Comment by ghost** [comment]
Same result:
https://github.com/user-attachments/assets/d6928c8f-3c96-4b0a-a9d7-bccafcb054ff

**Comment by stephenhawk8054** [comment]
Can you test these, one by one:

```adb
*$doc,replace=/popunder//,to=xhamster.*
xhamster.*##+js(aeld, click, [native code], elements, .main-wrap)
xhamster.*##+js(set-cookie, pShow, true)
```

**Comment by ghost** [comment]
All of them worked, no more popups.

However I noticed another issue, if you search for something, there are some spaces that continously load, likely ads (it happened even before and after testing the filters):
https://github.com/user-attachments/assets/776ea2bb-0e43-444c-82ee-1674a6bf81ab

**Comment by stephenhawk8054** [comment]
Where are the spaces?

**Comment by ghost** [comment]
When you search something using the search bar, among the results there are these boxes or spaces like in the screenshot above that have a loading animation, but they can't load anything.

**Comment by ghost** [comment]
https://github.com/user-attachments/assets/b4e1963f-06ef-4fc1-8ef0-4b9d313f6954

Like this

**Comment by dportvine** [comment]
`https://xhamster.com/search/erotica`

<details>
<summary>Screenshot(s)</summary>




![Image](https://github.com/user-attachments/assets/b0a57234-e4be-4633-a119-00fb6248f81f)



</details>


<details>

```yaml
uBlock Origin: 1.65.0
Firefox Mobile: 143
filterset (summary):
 network: 179078
 cosmetic: 117318
 scriptlet: 33333
 html: 2682
listset (total-discarded, last-updated):
 removed:
  adguard-mobile: null
  UKR-0: null
 added:
  https://easylist-downloads.adblockplus.org/bitblock.txt: 24508-22, now
  adguard-cookies: 34741-327, now
  ublock-cookies-adguard: 5070-74, now
  ublock-annoyances: 4559-41, 22m Δ
  RUS-0: 41412-243, 22m Δ
  RUS-1: 1823-254, now
  ublock-experimental: 11-0, 22m Δ
 default:
  user-filters: 0-0, never
  ublock-filters: 40880-86, 22m Δ
  ublock-badware: 11434-18, 22m Δ
  ublock-privacy: 3345-3, 22m Δ
  ublock-unbreak: 2730-2, 22m Δ
  ublock-quick-fixes: 288-13, 22m Δ
  easylist: 75312-217, 22m Δ
  easyprivacy: 54342-146, 22m Δ
  urlhaus-1: 31030-0, now
  plowe-0: 3419-899, now
filterset (user): [empty]
trustedset:
 added: [array of 1 redacted]
userSettings:
 webrtcIPAddressHidden: true
hiddenSettings:
 filterAuthorMode: true
 popupPanelHeightMode: 1
supportStats:
 allReadyAfter: 730 ms (selfie)
 maxAssetCacheWait: 231 ms
 cacheBackend: indexedDB
popupPanel:
 blocked: 11
 network:
  xhamster.com: 9
  googletagmanager.com: 1
  tsyndicate.com: 1
 extended:
  ##div[class*="cookiesAnnounce-"]
  ##.thumb-list-mobile-item--widget
  ##div[class$="-ts-init"]
  ##div[class*="cam-wgt"]
  ##div[data-role="promo-messages-wrapper"]
  ##+js(set-constant, initials.layout.layoutPromoProps.promoMessag…
  ##+js(remove-class, xh-thumb-disabled, html)
  ##+js(remove-cookie, video_view_count)
  ##+js(addEventListener-defuser, click, [native code], elements, …
  ##+js(trusted-replace-node-text, script, popunder, '', condition…
  ##+js(set-cookie, ts_popunder, true, '', reload, 1)
  ##+js(set-constant, initials.yld-pdpopunder, '')
```
</details>

**Comment by ghost** [comment]
Can confirm it does also happen on my side, and also when you search something and tap on the video, while tapping all the results are covered by a blue semi-transparent background, usually this happens to pop-ups.

**Comment by stephenhawk8054** [comment]
> when you search something and tap on the video, while tapping all the results are covered by a blue semi-transparent background, usually this happens to pop-ups.

I don't understand how to reproduce this issue.

**Comment by ghost** [comment]
Basically go to the page and search for something. When you tap on a video, in that period while you press, the entire space that contains all the videos becomes transparently blue (which I noticed for this to happen usually when a pop-up gets triggered), like this:

https://github.com/user-attachments/assets/5163857f-caa6-44d1-9e53-48fb8a00b871

Before, only the video I was clicking got covered by this blue background, not the entire space.

The blue background appears because it's something specific in Chromium browsers, on Firefox it doesn't appear.


Note: If this doesn't occur when searching for something, try tapping on a video and then go down to the list of suggested videos and tap on one of those.

**Comment by ghost** [comment]
> `https://xhamster.com/search/erotica`
> 
> Screenshot(s)

I also get the same section on Firefox mobile with uBO on.

**Comment by dportvine** [comment]
This is still actuall. 

`https://xhamster.com/search/erotica`

<details>
<summary>Screenshot(s)</summary>




![Image](https://github.com/user-attachments/assets/b0a57234-e4be-4633-a119-00fb6248f81f)



</details>

**Comment by ghost** [comment]
On my side it doesn't happen with Edge, but with Firefox it does.

However I think it counts as self-promotion since it's one of their sister sites.

**Comment by mapx-** [comment]
yeah, self promotion

add as custom filter

```adb
xhamster.*##[class^="mobileHorizontal"]:has([href^="https://xhamsterlive.com"])
```

**Comment by playonevideo** [comment]
When you play a video there is a massive spam of `go.hpyjmp.com` requests every second that slows down the page a lot and makes the browser slow too:

<details>

<img width="1609" height="895" alt="Image" src="https://github.com/user-attachments/assets/492e1957-97b0-4f0f-babd-03d6864c86b7" />

</details>

Video link for test: `https://xhamster.com/videos/i-suck-cock-for-her-xh6v3Wi`

After 10 seconds there are like 800+ requests blocked on the uBO icon.

**Comment by playonevideo** [comment]
The play button also has a noticeable delay when clicked on the player, likely due to the spam above.

**Comment by stephenhawk8054** [comment]
I can't reproduce.

**Comment by playonevideo** [comment]
How many requests get blocked on your side?

Mine already spammed 768 requests in just 10 seconds:

https://github.com/user-attachments/assets/32cfa272-18ea-4d8e-9b6a-f8e9eea20feb [NSFW]

Every second they spam like 60-70 requests from that domain.

**Comment by stephenhawk8054** [comment]
Mine is stable at 8. I don't see that domain appears:

<details>

<img width="3566" height="1749" alt="Image" src="https://github.com/user-attachments/assets/42708a7e-c1ff-43be-bf60-e345b9f2cce2" />

</details>

**Comment by playonevideo** [comment]
The site is already in EasyList, and xhamster is the one that massively spams it each second:

<details>

<img width="1590" height="826" alt="Image" src="https://github.com/user-attachments/assets/71b028db-b855-4818-b253-30c345acd051" />

</details>

Might be something regional if you can't reproduce. Try with a RO IP.

**Comment by stephenhawk8054** [comment]
Yeah, I mean I can't see that request fired on my side. Tried with VPN to Romania too.

**Comment by playonevideo** [comment]
Are there any other steps I can try?

**Comment by playonevideo** [comment]
https://github.com/uBlockOrigin/uAssets/issues/23538#issuecomment-5306904084 is now fixed by https://github.com/uBlockOrigin/uAssets/commit/ef8369943465100dea41decc9a531c6d2757bb54

**Comment by playonevideo** [comment]
From what I see, they are frequently changing the ad server domains, I already reported two of them at EasyList.

<img width="1584" height="898" alt="Image" src="https://github.com/user-attachments/assets/14b2156f-efb4-46d0-870b-545fd8cb98c0" />

Is there some way to stop the site from connecting to any such servers or they should be reported each one manually to EL?

**Comment by stephenhawk8054** [comment]
Maybe. But every filter can be countered by them. I don't have time to check the site now.

**Comment by playonevideo** [comment]
From what I notice, everytime I reported one of these ad servers to EL, they switched to a different one, so yeah, they are watching for sure.

**Comment by stephenhawk8054** [comment]
I can't reproduce. Please share exact video link and which country you are when accessing the site.

**Comment by playonevideo** [comment]
Here's a test link, on this one I noticed the ad server: `https://xhamster.com/videos/hetero-viejas-fotos-y-videos-de-maduros-y-abuelos-10-xhAt7FO`

I'm Romanian (it's visible on my profile as well, in case you probably didn't notice.)

**Comment by stephenhawk8054** [comment]
Ok, I just reproduced when using VPN to US.

**Comment by playonevideo** [comment]
https://github.com/easylist/easylist/pull/25190
